Trinity.Storage.Composite refactoring

yatli opened the pull request:
@thautwarm please ignore the data importer part. I'm migrating it to leverage `Trinity.Storage.Composite` instead of the old implementation of code building.

This PR updates `Trinity.Storage.Composite`:

1. namespace aligned to `Trinity.Storage.Composite`
2. references Trinity.TSL.CodeGen.exe, which will be available at runtime. thus we do not have to use a python collector script to collect the assemblies. just invoke "dotnet" and install the packages.
3. merge GraphEngine.DataImporter and its CoreCLR counterpart. Multi-targeting is achieved.
4. reduce accessibility levels for most classes in `Trinity.Storage.Composite`.
5. Simplify the setup of `Trinity.Storage.Composite` -- no more parameterized initial setup

Commit:
data import: refactoring wip

Commit:
GraphEngine.Storage.Composite: refactoring. Remove settings. Needs update from the python side

Commit:
GraphEngine.Storage.Composite: Utils.Session now tries to backup & restore data.

Commit:
merge python nupkg and python wheel package

Commit:
Trinity.FFI.Python: cleanup

Commit:
...

Commit:
Trinity.FFI.Python: restore dependencies automatically from nuget package root

Commit:
Python.FFI: add json.net as dependency

thautwarm reviewed (approved):
good job to fix typo~

thautwarm reviewed (approved):
Nice tao, you've made a very good use of `Utils.Session`! Also, thanks a lot for your refactoring.

Commit:
Trinity.FFI: refactoring

Commit:
Trinity.FFI.Python: build script improvements

Commit:
Trinity.FFI: auto generate Native.cs, FFIMethods.cs and the C++ counterpart, based on FFIExport attributes

Commit:
FFI: auto generate cellops and cellaccessorops

Commit:
minor touch

Commit:
follow nuget packaging conventions on native assemblies.

- the coreclr native assemblies (trinity.dll, libtrinity.so) are packed
into runtimes/{rid}/native
- the netfx native assembly (trinity.c.dll) uses existing assembly
release mechanism.
- note that a hosting program (the python GE package, started from
python side, for example) should take care of the native assemblies, as
we are not starting from the default host (dotnet.exe), which will
properly setup native assembly probe path according to the assets.

Commit:
Trinity.Core: do not distribute netfx Trinity.C to runtimes folder

Commit:
Storage.Composite: fix codegen lookup

Commit:
Composite: refactoring

Commit:
Composite: properly shadow copy assemblies; more robust assembly building; simplify persistent data

Commit:
Composite: more tests

Commit:
Composite: more robust composing
